### PR TITLE
refactor

### PR DESCRIPTION
**Pull request by rinckm**
### Description

refactoring only

### Submitter checklist

- [ ] Change has been tested (on a back-end cluster)
- [ ] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [ ] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [ ] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: pkg/runctl/controller.go [hunk 1]
```diff
@@ -576,6 +576,22 @@ func (c *Controller) handlePipelineRunWaiting(
 	return false, nil
 }
 
+func (c *Controller) startPipelineRun(ctx context.Context,
+	runManager run.Manager,
+	pipelineRun k8s.PipelineRun,
+	pipelineRunsConfig *cfg.PipelineRunsConfigStruct) (bool, error) {
```
Comment (anbrsap) [inline pkg/runctl/controller.go:582]:
```suggestion
	pipelineRunsConfig *cfg.PipelineRunsConfigStruct,
) (bool, error) {
```

Comment (anbrsap) [inline pkg/runctl/controller.go:582]: The `bool` return type is not required.

File: pkg/runctl/controller.go [hunk 2]
```diff
@@ -38,6 +38,8 @@ const (
 	// It is an invalid Kubernetes name to avoid conflicts with real
 	// pipeline runs.
 	heartbeatStimulusKey = "Heartbeat Stimulus"
+
+	waitingFailed = "waiting failed"
```
Comment (anbrsap) [inline pkg/runctl/controller.go:42]: The name of the constant doesn't reflect what it is about. Is it an error messare, a key, ...?

File: pkg/runctl/controller.go [hunk 3]
```diff
@@ -301,22 +303,12 @@ func (c *Controller) syncHandler(key string) error {
...
-		return err
-	}
-
-	err = c.handlePipelineRunAbort(ctx, pipelineRun)
-	if err != nil {
-		return err
-	}
-
-	err = c.handlePipelineRunResultExistsButNotCleaned(ctx, pipelineRun)
+	err = c.handleFastExit(ctx, pipelineRun)
```
[the remark this answers is not in the document]
Comment (rinckm) [reply pkg/runctl/controller.go:306]: I only found a way to exclude a block from all scans: https://docs.sonarqube.org/latest/project-administration/narrowing-the-focus/#ignoring-blocks-within-files

File: pkg/runctl/controller.go [hunk 4]
```diff
@@ -209,6 +211,11 @@ func (c *Controller) processNextWorkItem() bool {
 			return nil
 		}
 
+		if key == heartbeatStimulusKey {
+			c.heartbeat()
+			return nil
+		}
+
```
[the remark this answers is not in the document]
Comment (rinckm) [reply pkg/runctl/controller.go]: done

File: pkg/runctl/controller.go (hunk 2 above)
Comment (rinckm) [reply pkg/runctl/controller.go]: done

File: pkg/runctl/controller.go (hunk 1 above)
Comment (rinckm) [reply pkg/runctl/controller.go]: done

**Review by valiparsa — APPROVED** [review]

**Pull request merged**